dkc/core/models/folder.py

```diff
@@ -82,6 +84,25 @@ def ancestors(self) -> models.query.RawQuerySet[Folder]:
             [self.pk, Folder.MAX_TREE_HEIGHT + 1],
         )
 
+    @transaction.atomic
+    def increment_size(self, amount: int) -> None:
+        """
+        Increment or decrement the Folder size.
+
+        This function increments (or decrements, if ``amount`` is negative) the Folder's current
+        size.
+
+        If this increment would exceed the max allotment for this quota, a ``ValidationError``
+        is raised and the transaction is rolled back.
+        """
+        ancestor_pks = [folder.pk for folder in self.ancestors]
+
+        Folder.objects.filter(pk__in=ancestor_pks).update(used=(models.F('size') + amount))
+
+        # Update with the new value
+        # TODO: Try without this, test if the parent is cached
+        # self.refresh_from_db(fields=['used'])
```

Comment:
I think we'll need this, right?

Reply:
Or better yet, can we somehow invalidate the whole folder object without hitting the database, such that it'll be lazily refreshed if anyone tries to access attributes of it?

Reply:
In the case of a simple field (like `size` here), I *believe* doing `del folder.size` is sufficient to make the field deferred, such that it will be feteched lazily on future access, the same as if it had been fetched initially [using `.defer`](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#defer).

I'm not able to find a clean, easy way to invalidate / re-defer foreign key references after they've been resolved.

However, as far as I can find, deferring fields directly on a model like this after instantiation is not documented by Django and not officially supported. In a case where we really need it for performance, I'd be open to the idea, but for now, I think we'd be deviating from standard Django usage in the name of premature optimization.